src/diagnostic/detail/types/fluid.h

```diff
@@ -60,13 +90,7 @@ void FluidDiagnosticWriter<HighFiveDiagnostic>::getDataSetInfo(DiagnosticPropert
     auto& ions = hi5.modelView().getIons();
     std::string lvlPatchID{std::to_string(iLevel) + "_" + patchID};
 
-    auto checkActive = [&](auto& tree, auto var) {
-        auto active = tree + var;
-        bool b      = diagnostic.quantity == active;
-        if (b && !fileData.count(diagnostic.quantity))
-            fileData.emplace(diagnostic.quantity, hi5.makeFile(diagnostic));
-        return b;
-    };
+    auto checkActive = [&](auto& tree, auto var) { return diagnostic.quantity == tree + var; };
```

Comment:
shouldn't we generally capture only the one variable that's used in the lambda ? less source of (w)error (with shadowing etc.)

Reply:
> shouldn't we...

I don't see much benefit to this. I like [&] because it's minimalist and you get what you need.
If you want to choose an approach, maybe a style guide should exist

Reply:
ok I didn't find anything in the cpp core guidelines regarding how to capture things in lambdas.
My argument is that explicit is often better than implicit and when there is only one variable to capture, its explicit that it is *that one* that we want and no other.
But it's a discussion...